### ISSUE TITLE
Bug: invocation-type: Event is not supported. RequestResponse is only supported.

### ISSUE THREAD

**Issue by ijtarano** [aws/aws-sam-cli#6900]
### Description:
It seems like RequestResponse in boto3 invoke method doesn't work with SAM Local even when boto3 accept this parameter (https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/lambda/client/invoke.html)

### Steps to reproduce:
```python 
import boto3
import json 

payload = {
}

lambda_client = boto3.client("lambda", endpoint_url="http://localhost:3001", region_name="eu-west-1")
lambda_function_name = "Lambda"

payload_json = json.dumps(payload)

response = lambda_client.invoke(
    InvocationType="Event", # This Event fails
    FunctionName=lambda_function_name,
    Payload=payload_json
)

```

### Observed result:
ClientError: An error occurred (NotImplemented) when calling the Invoke operation: invocation-type: Event is not supported. RequestResponse is only supported.

### Additional environment details (Ex: Windows, Mac, Amazon Linux etc)

1. Amazon Linux:
2. `sam --version`: SAM CLI, version 1.113.0
3. AWS region: eu-west-1

```
{
  "version": "1.113.0",
  "system": {
    "python": "3.11.3",
    "os": "Linux-4.14.320-243.544.amzn2.x86_64-x86_64-with-glibc2.26"
  },
  "additional_dependencies": {
    "docker_engine": "20.10.23",
    "aws_cdk": "Not available",
    "terraform": "1.7.2"
  },
  "available_beta_feature_env_vars": [
    "SAM_CLI_BETA_FEATURES",
    "SAM_CLI_BETA_BUILD_PERFORMANCE",
    "SAM_CLI_BETA_TERRAFORM_SUPPORT",
    "SAM_CLI_BETA_RUST_CARGO_LAMBDA"
  ]
}
```

**Comment by ijtarano** [comment]
Yes, of course, It works perfectly with "RequestResponse" parameter ;)

**Comment by lucashuy** [comment]
Thanks for letting us know, this is something that we guard against since async execution isn't something that is currently implemented. I can move this issue to a feature request instead.

As an alternative, it's possible to use our `sam sync --watch` command to deploy to AWS and update code live, similar to how one would use `sam local start-lambda`. This could be a workaround for now.

**Comment by ericbambach-qwickrate** [comment]
Bump - this feature is a use-case for us and we want to test async events locally. As a workaround we use RequestResponse but its not ideal.

**Comment by Danny-Smart** [comment]
I also have this use case and for me, neither using "RequestResponse" nor `sam sync --watch` is a valid workaround. 

Searching through issues, I can see that this has been a problem since at least 2018 (https://github.com/aws/aws-sam-cli/issues/510#issuecomment-451268559) and has had at least one attempt at a solution closed off (#749)

Please can someone implement this feature?

**Comment by reedham-aws** [comment]
Hello, I believe this issue was fixed by #8590. It hasn't been released yet, although it will probably be available in version 1.154.0. I will update this issue when that is done.

**Comment by reedham-aws** [comment]
The PR has been merged, and I will close this issue.
